clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java

```diff
@@ -149,24 +149,28 @@ public boolean equals(Object o) {
 
         ConfigEntry that = (ConfigEntry) o;
 
-        return this.name.equals(that.name) &&
-                this.value != null ? this.value.equals(that.value) : that.value == null &&
```

Comment:
This was a bug as it evaluated following conditions:
```java
that.value == null &&
                this.isSensitive == that.isSensitive &&
                this.isReadOnly == that.isReadOnly &&
                this.source == that.source &&
                Objects.equals(this.synonyms, that.synonyms);
```
and the expected behavior is `that.value == null`.

There are some tests depending on the "incorrect" equals.

Reply:
@ijuma @dajac I filed a Jira for this PR as it includes a bug fix. Personally, it should be backport to all active branches. WDYT?